src/items.rs

```diff
@@ -1056,7 +1057,6 @@ pub fn rewrite_associated_type(ident: ast::Ident,
         let bounds: &[_] = &ty_param_bounds;
         let bound_str = bounds.iter()
             .filter_map(|ty_bound| ty_bound.rewrite(context, context.config.max_width, indent))
-            .collect::<Vec<String>>()
```

Comment:
Actually, I think this was a bug! Could you refactor this into an intersperse -> option collect -> try_opt!, like you did with the others?